planner/filter/external_conditions.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+package filter
+
+import (
+	"github.com/sourcenetwork/defradb/connor"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/planner/mapper"
+)
+
+// BuildExternalConditions builds a map of external conditions from the provided conditions.
+func BuildExternalConditions(conditions map[connor.FilterKey]any, mapping *core.DocumentMapping) map[string]any {
```

Comment:
suggestion: This looks like it is dead and could be removed?

Reply:
I've actually added this, because I need this for secondary indexes. But I can add it in secondary indexes branch. No problem